### ISSUE TITLE
IDE0018 removes necessary nullable annotations

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#38554]
**Version Used**: 
3.4.0-beta1-19426-03+b18d90790da95b6862def56802fa4249e4868bb4

**Steps to Reproduce**:
```C#
#nullable enable
class Program
{
    static void Main()
    {
        string? result;
        TryGetValue("hi", out result);
    }

    static bool TryGetValue(string key, out string? result)
    {
        result = null;
        return false;
    }
}
``` 

**Expected Behavior**:
IDE0018 offered and changes:
```C#
        string? result;
        TryGetValue("hi", out result);
```
into
```C#
        TryGetValue("hi", out string? result);
```

**Actual Behavior**:
IDE0018 offered and changes:
```C#
        string? result;
        TryGetValue("hi", out result);
```
into
```C#
        TryGetValue("hi", out string result);
```
which fails to compile with "warning CS8600: Converting null literal or possible null value to non-nullable type.".

**Comment by jasonmalinowski** [comment]
Appears to be fixed when testing with https://github.com/dotnet/roslyn/pull/40030. This is likely because the underlying compiler wrapping is now working better, but we'll close it when that merges.

**Comment by jasonmalinowski** [comment]
Confirmed fixed on latest 16.5 Preview 2 builds.
